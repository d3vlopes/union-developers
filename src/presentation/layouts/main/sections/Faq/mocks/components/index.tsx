import { vitest } from 'vitest'

vitest.mock('@/presentation/components/organisms', () => ({
  QuestionsAccordion: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return (
      <div data-testid="mock-questions-accordion">{children}</div>
    )
  },
}))
