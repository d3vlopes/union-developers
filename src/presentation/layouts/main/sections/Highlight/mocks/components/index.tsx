import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  Highlights: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return <div data-testid="mock-highlights">{children}</div>
  },
}))
