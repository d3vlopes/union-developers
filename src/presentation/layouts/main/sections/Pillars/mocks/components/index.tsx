import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  MethodologyCards: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return <div data-testid="mock-methodology-cards">{children}</div>
  },
}))
