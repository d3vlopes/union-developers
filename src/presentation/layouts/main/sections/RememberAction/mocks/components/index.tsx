import { vitest } from 'vitest'

vitest.mock(
  '@/presentation/components/molecules/RememberAction',
  () => ({
    RememberAction: function Mock({
      children,
    }: {
      children: React.ReactNode
    }) {
      return <div data-testid="mock-remember-action">{children}</div>
    },
  }),
)
