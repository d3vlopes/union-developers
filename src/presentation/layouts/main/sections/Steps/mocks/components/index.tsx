import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  Steps: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="mock-steps">{children}</div>
  },
}))
