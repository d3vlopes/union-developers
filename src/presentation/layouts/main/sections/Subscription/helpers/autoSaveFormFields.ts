import { Dispatch, SetStateAction } from 'react'

export function autoSaveFormFields<TFormValues, TFormData>(
  formValues: TFormValues,
  formData: TFormData,
  errors: object,
  setFormData: Dispatch<SetStateAction<TFormData>>,
  delay = 5000,
) {
  const interval = setInterval(() => {
    const isFieldContainError = Object.keys(errors).length > 0

    if (!isFieldContainError) {
      setFormData({ ...formData, ...formValues })
    }
  }, delay)

  return interval
}
