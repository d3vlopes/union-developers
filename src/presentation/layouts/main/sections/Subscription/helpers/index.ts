export * from './autoSaveFormFields'
export * from './formStepMap'
