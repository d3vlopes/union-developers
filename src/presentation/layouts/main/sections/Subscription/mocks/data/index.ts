import { generateId } from '@/utils'

import { termsMock } from '../../steps/AboutYou/components/Terms/mock'

import { SubscriptionSectionProps } from '../..'

export const subscriptionMock: SubscriptionSectionProps = {
  id: 'formulario',
  title: 'Junte-se ao time!',
  initialStepDescription:
    'Tem interesse em participar do Union? Confira abaixo os requisitos para participar',
  formStepsDescription:
    'Ficamos felizes em saber que você tem interesse em fazer parte do Union! Agora é só responder o nosso formulário',
  requirements: [
    'Ter mais que 18 anos',
    'Ter 1 ano ou mais de estudo em programação',
    'Ainda não trabalhar na área',
    'Já ter criado projetos sozinho que não sejam de cursos ou eventos',
    'Não participar de bootcamps ou programas de capacitação',
    'Ter disponibilidade para trabalhar no mínimo 1 hora por dia no projeto',
    'Ter disponibilidade para participar das reuniões do time, durante a semana e finais de semana',
  ],
  buttonText: 'Inscrever-se',
  formStepNavLinks: [
    {
      id: generateId(),
      stepTitle: 'Dados pessoais',
      stepNumber: 1,
    },
    {
      id: generateId(),
      stepTitle: 'Informações técnicas',
      stepNumber: 2,
    },
    {
      id: generateId(),
      stepTitle: 'Sobre você',
      stepNumber: 3,
    },
    {
      id: generateId(),
      stepTitle: 'Enviar',
      stepNumber: 4,
    },
  ],
  terms: termsMock,
}
