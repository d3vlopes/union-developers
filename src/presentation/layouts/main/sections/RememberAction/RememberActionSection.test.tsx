import { render, screen } from '@/__tests__/helpers'

import './mocks'

import { mainLayoutMock } from '@/presentation/layouts/main/mocks/data'

import { RememberActionSection } from '.'

const { getByTestId } = screen

const mock = mainLayoutMock.rememberActionSection

describe('<RememberActionSection />', () => {
  it('should render RememberAction', () => {
    render(<RememberActionSection {...mock} />)

    const rememberAction = getByTestId('mock-remember-action')

    expect(rememberAction).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<RememberActionSection {...mock} />)

    expect(container).toMatchSnapshot()
  })
})
