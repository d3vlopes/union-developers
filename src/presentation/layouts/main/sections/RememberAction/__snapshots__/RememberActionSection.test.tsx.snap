// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RememberActionSection /> > should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 1.3rem;
  padding: 8.5rem 1.6rem 5rem;
  width: 100%;
  height: 84.4rem;
}

@media (min-width:1400px) {
  .c0 {
    margin-top: 15.6rem;
    height: 53.9rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      data-testid="mock-remember-action"
    />
  </div>
</div>
`;
